api/groups_test.go

```diff
@@ -2351,5 +2352,324 @@ func TestAddUserToGroup(t *testing.T) {
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					awsErrCode := "InternalErrorException"
+					awsErrMessage := internalErrorDescription
+					awsOrigErr := errors.New(awsErrCode)
+					awsErr := awserr.New(awsErrCode, awsErrMessage, awsOrigErr)
+					return nil, awsErr
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					l, _ := strconv.Atoi(string(*input.GroupName)[len(*input.GroupName)-1:])
+					return ListGroupsUsers(l), nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					So(successResponse, ShouldBeNil)
+					So(errorResponse, ShouldNotBeNil)
+				},
+			},
+			{
+				"500 response users for group ",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					output := listGroups(1)
+					return &output, nil
+
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					awsErrCode := "InternalErrorException"
+					awsErrMessage := internalErrorDescription
+					awsOrigErr := errors.New(awsErrCode)
+					awsErr := awserr.New(awsErrCode, awsErrMessage, awsOrigErr)
+					return nil, awsErr
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					So(successResponse, ShouldBeNil)
+					So(errorResponse, ShouldNotBeNil)
+				},
+			},
+		}
+		for _, tt := range listGroupsUsers {
+			Convey(tt.description, func() {
+				m.ListGroupsFunc = tt.getGroupsFunc
+				m.ListUsersInGroupFunc = tt.listUsersForGroupFunc
+				r := httptest.NewRequest(http.MethodGet, getGroupsReportEndPoint, nil)
+				r.Header.Set("Accept", "text/csv")
+				urlVars := map[string]string{
+					"id": "efgh5678",
+				}
+				r = mux.SetURLVars(r, urlVars)
+				successResponse, errorResponse := api.ListGroupsUsersHandler(ctx, w, r)
+				tt.assertions(successResponse, errorResponse)
+			})
+		}
+	})
+}
+
+func listGroups(noOfGroups int) cognitoidentityprovider.ListGroupsOutput {
+	groupList := []*cognitoidentityprovider.GroupType{}
+	for i := 0; i < noOfGroups; i++ {
+		groupname := "group" + strconv.Itoa(i)
+		groupdescription := "group " + strconv.Itoa(i) + " description"
+		groups := cognitoidentityprovider.GroupType{
+			Description: &groupdescription,
+			GroupName:   &groupname,
+		}
+		groupList = append(groupList, &groups)
+	}
+	output := cognitoidentityprovider.ListGroupsOutput{
+		Groups:    groupList,
+		NextToken: nil,
+	}
+	return output
+}
+
+func ListGroupsUsers(noOfUsers int) *cognitoidentityprovider.ListUsersInGroupOutput {
+	userList := []*cognitoidentityprovider.UserType{}
+	var (
+		attributeEmail = "Email"
+	)
 
+	for i := 0; i < noOfUsers; i++ {
+		userAttributes := []*cognitoidentityprovider.AttributeType{}
+		userEmail := "user" + strconv.Itoa(i) + ".email@domain.test"
```

Comment:
fmt.Sprintf might be a better choice here.

Reply:
could combine it with the Username lower down as well. So then Username and Email are both using the same base pattern. E.g. 

Username: "user_0"
UserEmail: "user_0.email@domain.test"